### PR TITLE
[openssl] Fix static consumption

### LINKED ISSUE #19081: [openssl] windows static consumer link broken.
Consuming static triplets of openssl is broken on Windows after #18042.

**Environment**
- OS: Windows 10
- Compiler: MSVC 16.10.3

**To Reproduce**
1. Try to build a project against static openssl.

**Expected behavior**
For it to work like before #18042.

**Failure logs** 
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	LNK2001	unresolved external symbol __imp__CertOpenStore@20	MaxMain	D:\Plasma\CyanWorlds.comEngine\build\vc++2019-x86\Sources\Tools\MaxMain\libcrypto.lib(e_capi.obj)	1	
Error	LNK2001	unresolved external symbol __imp__CertCloseStore@8	MaxMain	D:\Plasma\CyanWorlds.comEngine\build\vc++2019-x86\Sources\Tools\MaxMain\libcrypto.lib(e_capi.obj)	1	
Error	LNK2001	unresolved external symbol __imp__CertEnumCertificatesInStore@8	MaxMain	D:\Plasma\CyanWorlds.comEngine\build\vc++2019-x86\Sources\Tools\MaxMain\libcrypto.lib(e_capi.obj)	1	
Error	LNK2001	unresolved external symbol __imp__CertFindCertificateInStore@24	MaxMain	D:\Plasma\CyanWorlds.comEngine\build\vc++2019-x86\Sources\Tools\MaxMain\libcrypto.lib(e_capi.obj)	1	
Error	LNK2001	unresolved external symbol __imp__CertDuplicateCertificateContext@4	MaxMain	D:\Plasma\CyanWorlds.comEngine\build\vc++2019-x86\Sources\Tools\MaxMain\libcrypto.lib(e_capi.obj)	1	
Error	LNK2001	unresolved external symbol __imp__CertFreeCertificateContext@4	MaxMain	D:\Plasma\CyanWorlds.comEngine\build\vc++2019-x86\Sources\Tools\MaxMain\libcrypto.lib(e_capi.obj)	1	
Error	LNK2001	unresolved external symbol __imp__CertGetCertificateContextProperty@16	MaxMain	D:\Plasma\CyanWorlds.comEngine\build\vc++2019-x86\Sources\Tools\MaxMain\libcrypto.lib(e_capi.obj)	1
```	

**Additional context**
PR in progress.

### PR DESCRIPTION
**Pull request by Hoikas**
Fixes consumers of static openssl who were broken by #18042 due to the introduction of a conflicting vcpkg-cmake-wrapper.

- #### What does your PR fix?  
    Fixes #19081

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
    NC

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
    Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
    Yes

### CODE REVIEW

File: ports/openssl/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -32,4 +32,34 @@ set(CMAKE_SYSTEM_IGNORE_PATH
 
 _find_package(${ARGS})
 
+if(OPENSSL_FOUND)
+    if("@OPENSSL_USE_WINDOWS_CMAKE_WRAPPER@")
+        list(APPEND OPENSSL_LIBRARIES Crypt32.lib ws2_32.lib)
+        if(TARGET OpenSSL::Crypto)
+            set_property(TARGET OpenSSL::Crypto APPEND PROPERTY INTERFACE_LINK_LIBRARIES "Crypt32.lib;ws2_32.lib")
```
Comment (Neumann-A) [inline ports/openssl/vcpkg-cmake-wrapper.cmake:39]: lowercase crypt32? I assume otherwise MinGW triplets will complain? cc @autoantwort @dg0yt ?

Comment (autoantwort) [reply ports/openssl/vcpkg-cmake-wrapper.cmake:39]: I have not tested it, but very likely it must be lowercase for mingw

Comment (Hoikas) [reply ports/openssl/vcpkg-cmake-wrapper.cmake:39]: MinGW triplets use the unix path.

**Comment by PhoebeHui** [comment]
See https://github.com/microsoft/vcpkg/pull/18475

**Comment by PhoebeHui** [comment]
@Hoikas, close this PR since PR #18475  has been merged.

**Pull request closed without merging**